### PR TITLE
Use only a single browser for multiple user profiles.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: pkg/pub_integration/lib/src/headless_env.dart [hunk 1]
```diff
@@ -9,43 +9,6 @@ import 'package:_pub_shared/validation/html/html_validation.dart';
...
-  Future<void> close() async {
-    while (_profiles.isNotEmpty) {
-      await _profiles.removeLast().close();
-    }
-  }
-}
-
 /// Creates and tracks the headless Chrome environment, its temp directories and
 /// and uncaught exceptions.
 class HeadlessEnv {
```
Comment (jonasfj) [inline pkg/pub_integration/lib/src/headless_env.dart:14]: I'd love it if we could give this class a descriptive name.

Maybe, Just `HeadlessBrowser` or just `TestBrowser` or something?

File: pkg/pub_integration/test/fake_sign_in_test.dart [hunk 2]
```diff
@@ -46,8 +46,8 @@ void main() {
       final browserSession = await fakeTestScenario.createAnonymousTestUser();
       String? firstSessionId;
       // sign-in page
-      await browserSession.withBrowserPage(
-        (page) async {
+      await browserSession.withBrowserPage((page) async {
+        {
           final rs = await page.gotoOrigin('/sign-in?fake-email=user@pub.dev');
```
Comment (jonasfj) [inline pkg/pub_integration/test/fake_sign_in_test.dart:51]: Isn't the idea that `withBrowserPage` will return a browser instance that is already signed-in for the given user?

File: pkg/pub_integration/test/fake_sign_in_test.dart [hunk 3]
```diff
@@ -46,8 +46,8 @@ void main() {
       final browserSession = await fakeTestScenario.createAnonymousTestUser();
```
Comment (jonasfj) [inline pkg/pub_integration/test/fake_sign_in_test.dart:46]: This is a `user` object, right? Sorry -- if the variable name confuses me :D

**Review by jonasfj — APPROVED** [review]

File: pkg/pub_integration/lib/src/headless_env.dart (hunk 1 above)
Comment (isoos) [reply pkg/pub_integration/lib/src/headless_env.dart:14]: Well, it should be called `BrowserWithCoverageTrackingAndCustomResponses`, but I would rather have a shorter name. `TestBrowser` is closer than just `HeadlessBrowser`, so renaming it for now...

**Pull request merged**